Source/Core/VideoCommon/ImGuiManager.cpp

```diff
@@ -0,0 +1,380 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <cinttypes>
+#include <sstream>
+
+#include "Common/MsgHandler.h"
+#include "Common/Timer.h"
+#include "Core/Config/NetplaySettings.h"
+#include "Core/ConfigManager.h"
+#include "Core/Movie.h"
+#include "VideoCommon/AbstractPipeline.h"
+#include "VideoCommon/AbstractShader.h"
+#include "VideoCommon/AbstractTexture.h"
+#include "VideoCommon/FPSCounter.h"
+#include "VideoCommon/ImGuiManager.h"
+#include "VideoCommon/NetPlayChatUI.h"
+#include "VideoCommon/NetPlayGolfUI.h"
+#include "VideoCommon/OnScreenDisplay.h"
+#include "VideoCommon/RenderBase.h"
+#include "VideoCommon/Statistics.h"
+#include "VideoCommon/VertexManagerBase.h"
+#include "VideoCommon/VertexShaderGen.h"
+
+#include "imgui.h"
+
+std::unique_ptr<VideoCommon::ImGuiManager> g_imgui_manager;
```

Comment:
Would this not belong within the renderer class instead of having separate global state for it?

Reply:
The long-term plan is to make the Renderer class more of a "GPU device" object. I know that currently it still has other stuff in there, but I've been slowly moving things out.

So, the UI-related data isn't really part of the GPU device, so IMO should be separate. If anything, it should be part of the overall "video backend" state. That said, things like netplay chat should be moved within the manager class, removing those globals.